### PR TITLE
VACMS-17426: Requires VA facility location when the event is using a VA facility location type

### PR DESCRIPTION
**Pull request by dsasser**
## Description

Forces an Event with a Facility location type to also specify the Facility location.

Relates to #17426

## Testing done
Manual and cypress

## Screenshots
Screenshot 2024-04-16 at 7 19 28 AM

## QA steps
As a content admin

1. Visit node/add/event
2. Set the 'Location type' to "At a VA facility"
   - [x] Validate that the 'Facility location' field is visible and required

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [x] Tests have been added if necessary.
- [x] Automated tests have passed.
- [x] Code Quality Tests have passed.
- [x] Acceptance Criteria in related issue are met.
- [x] Manual Code Review Approved.
- [x] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [x] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by omahane — CHANGES_REQUESTED** [review]
The fields seem to be required, having all the trappings of requirement, but I can save the page without filling in either VA location (when "At a VA facility" is chosen) or an address (when "At a non-VA location" is chosen).

I did this here: https://pr17859-vpvmby9elrebdlxtkz0lfeyzfggxlf4j.ci.cms.va.gov/node/67710/edit

Here's a video of me doing so.

https://www.loom.com/share/05c581e632e94fbf8786bb528fa6f9cf?sid=34adf16d-4c51-4c74-ae11-d071e7fdd5d4

**Comment by dsasser** [comment]
> The fields seem to be required, having all the trappings of requirement, but I can save the page without filling in either VA location (when "At a VA facility" is chosen) or an address (when "At a non-VA location" is chosen).
> 
> I did this here: https://pr17859-vpvmby9elrebdlxtkz0lfeyzfggxlf4j.ci.cms.va.gov/node/67710/edit
> 
> Here's a video of me doing so.
> 
> https://www.loom.com/share/05c581e632e94fbf8786bb528fa6f9cf?sid=34adf16d-4c51-4c74-ae11-d071e7fdd5d4

That's....not good at all. And I was able to replicate this prior to this change as well, so something very peculiar is going on that I'll need to look into.

**Comment by dsasser** [comment]
@edmund-dunn @omahane I verified that this is an existing bug, not introduced in this PR, and I think I have located the culprit: clientside validation (or so it seems).

Screenshot 2024-04-17 at 12 23 28 PM

If I disable clientside validation for all forms, then Events are validated client-side as expected. But re-enabling that setting, and I'm able to bypass client-side validation. To recreate on staging:

1. Visit node/67744/edit
2. Make no changes, and attempt to save (it shouldn't let you since a revision log is required).
3. Notice how it let you submit the form unexpectedly (no browser/client-side error), but thankfully Drupal's backend field validation prevented saving.
4. Visit /admin/config/user-interface/clientside-validation
5. Uncheck the box to disable clientside validation for all forms, and save the form.
6. Revisit node/67744/edit (make sure you load the page again if you have it open in a separate tab)
7. Try to save.
8. Notice you get the expected browser validation error

**Comment by jilladams** [comment]
Noting: next steps for what we do about that bug are in discussion on the ticket; https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17426#issuecomment-2062420134

**Comment by dsasser** [comment]
@omahane given we know that the [issue you found](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/17859#pullrequestreview-2003952724) is pre-existing, how would you feel about reviewing this with Clientside Validation turned off?

File: docroot/modules/custom/va_gov_events/js/eventFormHelpers.es6.js [hunk 1]
```diff
@@ -292,11 +295,22 @@
     }
   };
 
+  const toggleFacilityLocationRequiredFields = (enableDisable, addRemove) => {
```
Comment (omahane) [inline docroot/modules/custom/va_gov_events/js/eventFormHelpers.es6.js:298]: I am unsure why `enableDisable`. I see that it's used elsewhere:
`fieldAddressAdminArea.required = enableDisable;`

Is there a functional reason for the difference?

**Review by omahane — APPROVED** [review]
Code looks good. And the UI/UX is working. Let's hope that the subsequent work to get clientside_validation out of the way will make the browser validation work correctly.

**Pull request merged**